core/signal/trap_spec.rb

```diff
@@ -221,6 +221,25 @@
       Signal.trap(:HUP, @saved_trap).should equal(@proc)
     end
 
+    it "calls #to_str on an object to convert to a String" do
+      obj = mock("signal")
+      obj.should_receive(:to_str).exactly(2).times.and_return("HUP")
+      Signal.trap obj, @proc
+      Signal.trap(obj, @saved_trap).should equal(@proc)
+    end
+
+    it "accepts Integer values" do
+      hup = Signal.list["HUP"]
+      Signal.trap hup, @proc
+      Signal.trap(hup, @saved_trap).should equal(@proc)
+    end
```

Comment:
suggestion: I would check also that `Signal.list["HUP"]` returns Integer but not some other type:
```ruby
hup = Signal.list["HUP"]
hup.should.is_a?(Integer)
```

Reply:
It may help to maintain this test case in future (and make it more readable) as for me. As a reader I may not remember what `Signal.list` returns at all and explicit check just makes the value type explicit. But it's a subjective topic of course.